numpyro/distributions/kl.py

```diff
@@ -136,3 +156,46 @@ def kl_divergence(p, q):
     t3 = p.concentration - q.concentration
     t4 = digamma(p.concentration) - digamma(sum_p_concentration)[..., None]
     return t1 - t2 + (t3 * t4).sum(-1)
+
+
+@dispatch(Gamma, Gamma)
+def kl_divergence(p, q):
+    # From https://en.wikipedia.org/wiki/Gamma_distribution#Kullback%E2%80%93Leibler_divergence
+    a, b = p.concentration, p.rate
+    alpha, beta = q.concentration, q.rate
+    b_ratio = beta / b
+    t1 = gammaln(alpha) - gammaln(a)
+    t2 = (a - alpha) * digamma(a)
+    t3 = alpha * jnp.log(b_ratio)
+    t4 = a * (b_ratio - 1)
+    return t1 + t2 - t3 + t4
+
+
+@dispatch(Weibull, Gamma)
+def kl_divergence(p, q):
+    # From https://arxiv.org/abs/1401.6853 Formula (28)
+    a, b = p.concentration, p.scale
+    alpha, beta = q.concentration, q.rate
+    a_reciprocal = 1 / a
+    b_beta = b * beta
+    t1 = jnp.log(a) + gammaln(alpha)
+    t2 = alpha * (jnp.euler_gamma * a_reciprocal - jnp.log(b_beta))
+    t3 = b_beta * jnp.exp(gammaln(a_reciprocal + 1))
+    return t1 + t2 + t3 - (jnp.euler_gamma + 1)
+
+
+@dispatch(Kumaraswamy, Beta)
+def kl_divergence(p, q):
+    # From https://arxiv.org/abs/1605.06197 Formula (12)
+    a, b = p.concentration1, p.concentration0
+    alpha, beta = q.concentration1, q.concentration0
+    b_reciprocal = jnp.reciprocal(b)
+    a_b = a * b
+    t1 = (alpha / a - 1) * (jnp.euler_gamma + digamma(b) + b_reciprocal)
+    t2 = jnp.log(a_b) + betaln(alpha, beta) + (b_reciprocal - 1)
```

Comment:
I believe you are using -E[log p/q], from which I get `-jnp.log(a_b)-betaln(alpha,beta) + (b_reciprocal -1)`.

Reply:
Here I'm calculating E_p(log p / q) where p=Kuma(a, b), q=Beta(alpha,beta). I'm using the same formula as in the reference, i.e. same notations for a, b, alpha, beta, but swapping notations for p and q (i.e. p here is q in the reference, q here is p in the reference).

Reply:
Thanks for the clarification. Found the problem and the implementation is consistent with the paper.